enhancements/scheduling/load-aware-vm-rebalancing-through-descheduler.md

```diff
@@ -0,0 +1,472 @@
+    devMultiSoftTainting: false
+    # evict multiple pods per node during a descheduling cycle
+    devMultiEvictions: Simple
+```
+
+Given the new profile is VM oriented we can expose or extend `devLowNodeUtilizationThresholds` with a different/new predefined thresholds.
+
+To reduce the risk of excessive exposure, the new `devActualUtilizationProfile` field will accept only a predefined set of profiles, each corresponding to a specific Prometheus query. After the initial integration testing and customer feedback, the field can be updated to support a broader range of values.
+
+Additionally, the descheduler already includes the `useDeviationThresholds` withing the [LowNodeUtilization](https://github.com/kubernetes-sigs/descheduler/?tab=readme-ov-file#lownodeutilization) plugin, which alters how utilization thresholds are calculated. Exposing this option through the operator could enhance overall performance during VM rebalancing. This option adjusts the values of `devLowNodeUtilizationThresholds`, as the thresholds are interpreted as the distance from the average utilization, either lower or higher.
+
+To increase the leverage against over-utilized nodes, the taint controller can be configured to apply multiple soft taints based on overall resource pressure through `devMultiSoftTainting` option. For example, when a node is classified as over-utilized and its utilization exceeds 50% of the remaining capacity, a second soft taint could be applied. The scaling of taints can follow different models, such as linear or exponential, depending on the desired behavior.
+
+In certain situations, it may be beneficial to evict multiple pods per node within a single descheduling cycle. By exposing `devMultiEvictions` with predefied values of `Simple` (1), `Modest` (2) and `Rapid` (5), one can explore this direction further.
+
+Currently, there is no dependency between the descheduler and the scheduler, so no additional validation of either configuration is required. However, when the scheduler configuration is extended with load-aware elements, the descheduler operator is expected to scale the descheduler down to zero if an invalid combination of configurations is detected.
+
+### Workflow Description
+
+On the descheduler side:
+1. The node classification thresholds gets configured as static (using `thresholds` and `targetThresholds`, as configured in the current `LowNodeUtilization` plugin) or dynamic, based on the state of other nodes in the cluster (using the `useDeviationThresholds` option within the plugin).
+1. The `LowNodeUtilization` plugin classifies nodes into three categories: under-utilized, appropriately-utilized, and over-utilized, as it does today. The metric used for classification will be derived from a PromQL query executed against a Prometheus instance. Prometheus will handle the aggregation of historical data and time series processing. The query is expected to return a scalar [0-1] value for each node.
```

Comment:
Maybe it's worth to mention that we intend to use recording rules on Prometheus side to emphasize that we want keep aggregation of historical data and time series processing there keeping the descheduler simpler.

Reply:
Done.